### PR TITLE
feat(HNG-138) : build waitlist page

### LINKED ISSUE #138: [Feature]: Develop Form on the Waitlist Page to Add Users to the Waitlist
### Description

Develop a form component for the waitlist page that collects user information (such as name, email address, and optionally phone number) and integrates with backend API endpoints to add users to the waitlist. Ensure the form matches the design specifications, handles form validation, and provides a smooth user experience.

### Acceptance Criteria

- [x] Develop a form with fields for user's name, email address.
- [x] Implement form validation to ensure required fields are filled and valid email format is entered.
- [ ] Integrate the form with backend API endpoints to add users to the waitlist.
- [x] Display success or error messages based on API response.
- [x] Ensure the form matches the provided design specifications and is responsive.
- [x] Write unit tests to verify form validation, API integration, and responsiveness

**Deadline Friday 10am**
### Links

https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=518-7790&m=dev
image

### PR DESCRIPTION
**Pull request by G4EVA-dev**
# Description

**Closes #138**

# Changes proposed

## What were you told to do?

[Feature]: Develop Form on the Waitlist Page to Add Users to the Waitlist

## What did you do?

 - Develop a form with fields for user's name, email address.
 - Implement form validation to ensure required fields are filled and valid email format is entered.
 - Integrate the form with backend API endpoints to add users to the waitlist.
 - Display success or error messages based on API response.
 - Ensure the form matches the provided design specifications and is responsive.
 - Write unit tests to verify form validation, API integration, and responsiveness

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [] My code follows the code style of this project. [x]
- [] This PR does not contain plagiarized content. [x]
- [] The title and description of the PR is clear and explains the approach. [x]
- [] I am making a pull request against the **dev branch** (left side). [x]
- [] My commit messages styles matches our requested structure. [x]
- [] My code additions will fail neither code linting checks nor unit test. []
- [] I am only making changes to files I was requested to. [x]

# Screenshots/Videos

Screenshot from 2024-07-19 06-01-23
Screenshot from 2024-07-19 06-01-05
Screenshot from 2024-07-19 06-03-29
Screenshot from 2024-07-19 06-04-08

Link to Figma Design:
https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=518-7790&m=dev

### CODE REVIEW

**Comment by oreoluwa212** [comment]
Update your branch to match dev and make a PR asap please (your tailwind.config has conflict because we now have global config). ensure lint and test passes

**Review by oreoluwa212 — COMMENTED** [review]
Firstly, Update your branch to match dev and make a PR asap please. ensure lint and test passes. 
Also follow this convention for images in public > images > yourpagename
Attach screenshots to this effects

**Comment by G4EVA-dev** [comment]
Screenshot from 2024-07-20 10-33-53

The conflicts I'm facing right is as a resulting of someone's work tha I pulled.
None of my wirtten code is conflicting. someone elses' component that is conflicting

**Comment by G4EVA-dev** [comment]
After running pnpm test:ui 
Screenshot from 2024-07-20 12-18-58

After running `pnpm lint:fix`
The errors here are from other files. I don't know why I'm facing the error.
I ran `git merge upstream/dev`
Screenshot from 2024-07-20 11-54-55

After running `pnpm test` All test pased
Screenshot from 2024-07-20 12-22-57

**Comment by G4EVA-dev** [comment]
### Image File Structure
Screenshot from 2024-07-20 13-50-16

**Comment by G4EVA-dev** [comment]
It's up to date on my end.
The passSuccessfulModel component is giving me problems.
I don't know whoever is working on it 

Screenshot from 2024-07-20 14-04-48

**Comment by G4EVA-dev** [comment]
please check again

**Comment by G4EVA-dev** [comment]
@oreoluwa212 please check and let me know what to fix

File: html/assets/index-TpLatVz2.js
Comment (taiwonaf) [inline html/assets/index-TpLatVz2.js:1]: Pull form upstream/dev to update your gitignore. You are not supposed to push this

File: src/app/page.test.tsx [hunk 1]
```diff
@@ -5,7 +5,11 @@ describe("page tests", () => {
   it("should render correctly", () => {
     expect.assertions(1);
 
-    render(<Page />);
+    render(
+      <>
```
Comment (taiwonaf) [inline src/app/page.test.tsx]: remove this and put only the page

File: src/components/common/WaitListForm/UserForm.tsx [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+import React, { FormEvent, useState } from "react";
+
+interface FormData {
+  name: string;
+  email: string;
+}
+
+const UserForm: React.FC = () => {
+  const [formData, setFormData] = useState<FormData>({ name: "", email: "" });
+  const [errors, setErrors] = useState<{ name?: string; email?: string }>({});
```
Comment (taiwonaf) [inline src/components/common/WaitListForm/UserForm.tsx]: types is missing

File: src/components/common/WaitListForm/UserForm.tsx [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+  const [success, setSuccess] = useState(false);
```
Comment (taiwonaf) [inline src/components/common/WaitListForm/UserForm.tsx]: missing boolean type

File: src/components/common/WaitListForm/UserForm.tsx [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+  };
+
+  const handleSubmit = (event: FormEvent) => {
+    event.preventDefault();
+    const newErrors = validate();
+    if (Object.keys(newErrors).length > 0) {
+      setErrors(newErrors);
+    } else {
+      console.log("Form data submitted:", formData);
+      // Clear form and show success message
```
Comment (taiwonaf) [inline src/components/common/WaitListForm/UserForm.tsx]: remove comments

File: src/components/common/WaitListForm/UserForm.tsx [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+    return newErrors;
+  };
+
+  const handleSubmit = (event: FormEvent) => {
+    event.preventDefault();
+    const newErrors = validate();
+    if (Object.keys(newErrors).length > 0) {
+      setErrors(newErrors);
+    } else {
+      console.log("Form data submitted:", formData);
```
Comment (taiwonaf) [inline src/components/common/WaitListForm/UserForm.tsx]: remove console.log

**Review by taiwonaf — COMMENTED** [review]
Check the reviews

**Comment by G4EVA-dev** [comment]
### **Fix Lint error**
Screenshot from 2024-07-21 01-20-08

**Comment by oreoluwa212** [comment]
i dont want to see any html

**Comment by G4EVA-dev** [comment]
@oreoluwa212 please where's the html ?

**Comment by G4EVA-dev** [comment]
Good morning ma.
Please which HTML ?

**Comment by oreoluwa212** [comment]
`This branch has conflicts that must be resolved
 to resolve conflicts before continuing.
Conflicting files
html/html.meta.json.gz`

**Comment by G4EVA-dev** [comment]
should I delete it ?
I don't even know where it's coming from

**Comment by G4EVA-dev** [comment]
@oreoluwa212 please check again
sorry ma

**Comment by oreoluwa212** [comment]
you were asked to do 1 form?why ami seeing that thumbs up page ??? [issue](https://github.com/hngprojects/hng_boilerplate_nextjs/issues/138)

read it, compare


are you ready?

**Comment by G4EVA-dev** [comment]
@oreoluwa212 
I thought I was the one to do the thumbs up page.
So, should I remove it ?

**Comment by oreoluwa212** [comment]
You are not in sync with upstream, pull, and ensure you accept incoming for these files when merging 

`
This branch has conflicts that must be resolved
Conflicting files
pnpm-lock.yaml
src/app/(home)/page.tsx
`

**Comment by oreoluwa212** [comment]
This PR is probably old, I am seeing html.meta and all in your file changes and that caused us dearly.

Please pull upstream and accept incoming, make a new PR, attach updated tests screenshots too

**Comment by G4EVA-dev** [comment]
@oreoluwa212 
Attached is the screenshot for test and git from the dev.
I don't know why the html file keeps comimg 
Screenshot from 2024-07-22 02-44-33

**Pull request closed without merging**